tools/duplicate_function_eliminator.py

```diff
@@ -0,0 +1,428 @@
+
```

Comment:
minify_shell in retrospect is probably not a good choice of name. We need to run DFE on the asm shell since there may be function references inside arrays there. We need to replace those references with their replacement (if any). 

Hope this makes sense. Would eliminate_duplicate_funcs_on_shell be a better name?